config/config.go

```diff
@@ -544,6 +544,18 @@ type User struct {
 	// if omitted or zero - no limits would be applied
 	ReqPerMin uint32 `yaml:"requests_per_minute,omitempty"`
 
+	// The burst of summary upload packet size in a period of time(eg.10s) for user
+	// if omitted or zero - no limits would be applied
+	PacketSizeTokenLimitBurst int `yaml:"packet_size_token_limit_burst,omitempty"`
+
+	// packet size token produced rate limit
+	// if omitted or zero - no limits would be applied
+	PacketSizeTokenRate float64 `yaml:"packet_size_token_rate,omitempty"`
+
+	// packet size unit
+	// if omitted by empty - GB would be applied
+	PacketSizeUnit string `yaml:"packet_size_unit,omitempty"`
```

Comment:
have a dedicate field to specify the unit is inconsistent with the other variables in the conf. We're using only one variable to store the size and the unit using the ByteSize type:
https://github.com/ContentSquare/chproxy/blob/23e016a3bd918fc13b592a46a04261b6226ebefe/config/types.go#L21
You should look at other variable using a ByteSize to understand how to use it.

Reply:
i would consider it